### ISSUE TITLE
Faking mysql doesn't work

### ISSUE THREAD

**Issue by DanielSiepmann** [niklongstone/Fakerino#15]
Using the following call on CLI: `app/fake -t tx_downloadcounter_domain_model_download -n 100 -c configs/git.typo3.org.php` throws the following error:

```Fatal error: Call to a member function format() on string in /Volumes/Work/Projects/apps/fakerino/vendor/doctrine/dbal/lib/Doctrine/DBAL/Types/DateTimeType.php on line 53

Call Stack:
    0.3657     237568   1. {main}() /Volumes/Work/Projects/apps/fakerino/app/fake:0
    0.3697     616424   2. Fakerino\Core\Console\FakeConsole->run() /Volumes/Work/Projects/apps/fakerino/app/fake:30
    0.3769     907552   3. Fakerino\Core\FakeDataFactory->fakeTable(???) /Volumes/Work/Projects/apps/fakerino/src/Fakerino/Core/Console/FakeConsole.php:92
  212.1116    3122104   4. Fakerino\Core\Filler\DbFiller->fill() /Volumes/Work/Projects/apps/fakerino/src/Fakerino/Core/FakeDataFactory.php:122
  257.4617    3265192   5. Fakerino\Core\Database\DoctrineLayer->insert(???) /Volumes/Work/Projects/apps/fakerino/src/Fakerino/Core/Filler/DbFiller.php:78
  257.4639    3459432   6. Doctrine\DBAL\Connection->executeQuery(???, ???, ???, ???) /Volumes/Work/Projects/apps/fakerino/src/Fakerino/Core/Database/DoctrineLayer.php:125
  257.4654    3514504   7. Doctrine\DBAL\Connection->_bindTypedValues(???, ???, ???) /Volumes/Work/Projects/apps/fakerino/vendor/doctrine/dbal/lib/Doctrine/DBAL/Connection.php:846
  257.4657    3518448   8. Doctrine\DBAL\Connection->getBindingInfo(???, ???) /Volumes/Work/Projects/apps/fakerino/vendor/doctrine/dbal/lib/Doctrine/DBAL/Connection.php:1488
  257.4657    3518512   9. Doctrine\DBAL\Types\DateTimeType->convertToDatabaseValue(???, ???) /Volumes/Work/Projects/apps/fakerino/vendor/doctrine/dbal/lib/Doctrine/DBAL/Connection.php:1523
```

I've traced it down to the point, that Doctrine doesn't provide the length information and therefore data is truncated to 0.

I'm using PHP 5.6.29 (cli) and mysql 5.7.10.

Installation of fakerino was done through  `composer create-project fakerino/fakerino fakerino`

The configuration file looks like:
```
<?php

$conf['database'] = array(
    'dbname' => 'typo3_git_62',
    'user' => 'root',
    'password' => '',
    'host' => 'localhost',
    'driver' => 'pdo_mysql',
);
```

**Comment by DanielSiepmann** [comment]
Searching for the issue it looks like that is not possible at all: http://stackoverflow.com/questions/16591726/getting-the-length-of-a-column-with-doctrine-dbal

For some values it's hardcoded to `null`, see https://github.com/doctrine/dbal/blob/master/lib/Doctrine/DBAL/Schema/MySqlSchemaManager.php#L175

Also I don't see how faking date times should work using Doctrine with current code.

**Comment by niklongstone** [comment]
Thanks for reporting that.
Could you please tell me which typo3 version you are using or write the schema of that table so I can replicate the issue.

**Comment by DanielSiepmann** [comment]
Of course, the create statement is the following:
```
CREATE TABLE `tx_downloadcounter_domain_model_download` (
  `uid` int(11) NOT NULL AUTO_INCREMENT,
  `pid` int(11) NOT NULL DEFAULT '0',
  `name` varchar(255) NOT NULL DEFAULT '',
  `count` varchar(255) NOT NULL DEFAULT '',
  `date` datetime DEFAULT '0000-00-00 00:00:00',
  `page_title` varchar(255) NOT NULL DEFAULT '',
  `tstamp` int(11) unsigned NOT NULL DEFAULT '0',
  `crdate` int(11) unsigned NOT NULL DEFAULT '0',
  `cruser_id` int(11) unsigned NOT NULL DEFAULT '0',
  `deleted` tinyint(4) unsigned NOT NULL DEFAULT '0',
  `hidden` tinyint(4) unsigned NOT NULL DEFAULT '0',
  `starttime` int(11) unsigned NOT NULL DEFAULT '0',
  `endtime` int(11) unsigned NOT NULL DEFAULT '0',
  `t3ver_oid` int(11) NOT NULL DEFAULT '0',
  `t3ver_id` int(11) NOT NULL DEFAULT '0',
  `t3ver_wsid` int(11) NOT NULL DEFAULT '0',
  `t3ver_label` varchar(255) NOT NULL DEFAULT '',
  `t3ver_state` tinyint(4) NOT NULL DEFAULT '0',
  `t3ver_stage` int(11) NOT NULL DEFAULT '0',
  `t3ver_count` int(11) NOT NULL DEFAULT '0',
  `t3ver_tstamp` int(11) NOT NULL DEFAULT '0',
  `t3ver_move_id` int(11) NOT NULL DEFAULT '0',
  `sys_language_uid` int(11) NOT NULL DEFAULT '0',
  `l10n_parent` int(11) NOT NULL DEFAULT '0',
  `l10n_diffsource` mediumblob,
  PRIMARY KEY (`uid`),
  KEY `parent` (`pid`),
  KEY `t3ver_oid` (`t3ver_oid`,`t3ver_wsid`),
  KEY `language` (`l10n_parent`,`sys_language_uid`)
) ENGINE=InnoDB AUTO_INCREMENT=201 DEFAULT CHARSET=utf8;
```

The main reason is independent as far as I understood the Doctrine Code. As doctrine will reset the length for fields like ints and Fakerino will truncate the faked data to 0 in that case.

**Comment by niklongstone** [comment]
@DanielSiepmann thanks for reporting that.
I have applied a quick fix in a branch called `fix-doctrine-datetime`.
I would like to perform more test before to release that fix. 

In the meanwhile you can use that branch with: `composer create-project fakerino/fakerino:dev-fix-doctrine-datetime fakerino`

**Comment by niklongstone** [comment]
fixed in [0.8.3](https://github.com/Fakerino/Fakerino/releases/tag/0.8.3)
